controller/authorize_blackbox_test.go

```diff
@@ -0,0 +1,301 @@
+package controller_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/app/test"
+	"github.com/fabric8-services/fabric8-auth/client"
+	. "github.com/fabric8-services/fabric8-auth/controller"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	testsupport "github.com/fabric8-services/fabric8-auth/test"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/goadesign/goa"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestAuthorizeREST struct {
+	gormtestsupport.DBTestSuite
+}
+
+func TestRunAuthorizeREST(t *testing.T) {
+	suite.Run(t, &TestAuthorizeREST{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (rest *TestAuthorizeREST) UnSecuredController() (*goa.Service, *AuthorizeController) {
+	svc := testsupport.ServiceAsUser("Login-Service", testsupport.TestIdentity)
+	loginService := newTestKeycloakOAuthProvider(rest.Application)
+	return svc, &AuthorizeController{Controller: svc.NewController("AuthorizeController"), Auth: loginService, Configuration: rest.Configuration}
+}
+
+func (rest *TestAuthorizeREST) TestAuthorizeOK() {
+	t := rest.T()
+	svc, ctrl := rest.UnSecuredController()
+
+	redirect := "https://openshift.io"
+	clientID := rest.Configuration.GetPublicOauthClientID()
+	responseType := "code"
+	state := uuid.NewV4()
+
+	test.AuthorizeAuthorizeTemporaryRedirect(t, svc.Context, svc, ctrl, nil, clientID, redirect, responseType, nil, state)
+}
+
+func (rest *TestAuthorizeREST) TestAuthorizeBadRequest() {
+	t := rest.T()
+
+	rw := httptest.NewRecorder()
+	u := &url.URL{
+		Path: fmt.Sprintf(client.AuthorizeAuthorizePath()),
+	}
+	req, err := http.NewRequest("GET", u.String(), nil)
+	require.Nil(t, err)
+
+	redirectURI := "https://openshift.io/somepath"
+	prms := url.Values{}
+
+	prms.Add("response_type", "code")
+	prms.Add("redirect_uri", redirectURI)
+	prms.Add("client_id", rest.Configuration.GetPublicOauthClientID())
+	prms.Add("state", uuid.NewV4().String())
+
+	ctx := context.Background()
+
+	prms.Del("response_type")
+	goaCtx := goa.NewContext(goa.WithAction(ctx, "AuthorizeTest"), rw, req, prms)
+	authorizeCtx, err := app.NewAuthorizeAuthorizeContext(goaCtx, req, goa.New("LoginService"))
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "400 invalid_request: missing required parameter")
+	jsonapi.JSONErrorResponse(authorizeCtx, err)
+	require.Equal(t, 400, rw.Code)
+	prms.Add("response_type", "code")
```

Comment:
Lines 71-78 are mostly duplicated a few times for every param. Extract this code to a separate function and re-use it in the test for all params.